addon/index.js

```diff
@@ -58,8 +47,17 @@ export {
   getPretender,
 };
 
+/**
+ * Setup and teardown code, intended to be called with qunit hooks so that it can run code before & after each test.
+ */
 export function setupFactoryGuy(hooks) {
   hooks.beforeEach(function () {
-    manualSetup(this);
+    const owner =
+      this.owner || (this.container && this.container.owner) || this;
+    FactoryGuy.setStore(owner.lookup('service:store'));
+  });
+
+  hooks.afterEach(function () {
+    FactoryGuy.reset();
```

Comment:
Note that we're not calling `loadScenarios()` or `loadFactories()` anymore, avoiding `require()` altogether - it's up to the app to import what they want to use.

A migration guide or the changelog will detail the changes needed, but briefly;
* they'll want to import their factories before any tests run (likely in test-helper.js)
* they'll want to import any scenario(s) they want to use, and call run(), for each test (likely making use of beforeEach hooks)

See readme for more info

---

would have liked to use `getOwner(this)` for the owner code because simpler, but it doesn't seem available (guessing because ember 3.28? I think it was introduced in 4.x?). If we end up dropping ember 3.x I'll look into using it

Will look into the store service deprecations in later work